crypto/ec/ec2_smpl.c

```diff
@@ -603,9 +603,9 @@ int ec_GF2m_simple_is_on_curve(const EC_GROUP *group, const EC_POINT *point,
     if (!BN_GF2m_add(lh, lh, y2))
         goto err;
     ret = BN_is_zero(lh);
+
  err:
-    if (ctx)
-        BN_CTX_end(ctx);
+    BN_CTX_end(ctx);
     BN_CTX_free(new_ctx);
```

Comment:
One can wonder if there is advantage in `if (new_ctx != ctx) BN_CTX_free(new_ctx); else BN_CTX_end(ctx);`. I mean in case when ctx == new_ctx is there a point to "end" it prior freeing?

Reply:
Seen ! possible optimization, smart move ... but it will discard some debug information from CTXDBG_ENTRY/CTXDBG_EXIT calls when BN_CTX_DEBUG is enabled.

Reply:
Fair enough...